File: database/models/media_record.py
from sqlalchemy import Column, String, DateTime, Text
from datetime import datetime
from database.base import Base


class MediaRecord(Base):
    """存储媒体文件元数据"""
    __tablename__ = "media_record"

    media_id = Column(String(256), primary_key=True)
    source = Column(String(32), nullable=False, comment="媒体来源：upload / qq_chat / web_search / agent_save")
    original_source = Column(Text, nullable=False, comment="原始 URL 或路径")
    rustfs_path = Column(Text, nullable=False, comment="RustFS 保存路径")
    name = Column(String(512), nullable=True, comment="文件名")
    description = Column(Text, nullable=True, comment="图片文字描述")
    mime_type = Column(String(128), nullable=True, comment="MIME 类型，如 image/jpeg")
    created_at = Column(DateTime, nullable=False, default=datetime.utcnow, comment="记录创建时间")

File: zihuan_core/src/database/ddl/media_record.rs
pub const MYSQL_DDL: &str = "CREATE TABLE IF NOT EXISTS media_record (
        media_id VARCHAR(256) PRIMARY KEY,
        source VARCHAR(32) NOT NULL,
        original_source TEXT NOT NULL,
        rustfs_path TEXT NOT NULL,
        name VARCHAR(512),
        description TEXT,
        mime_type VARCHAR(128),
        created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP
    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4";

pub const SQLITE_DDL: &str = "CREATE TABLE IF NOT EXISTS media_record (
        media_id TEXT PRIMARY KEY,
        source TEXT NOT NULL,
        original_source TEXT NOT NULL,
        rustfs_path TEXT NOT NULL,
        name TEXT,
        description TEXT,
        mime_type TEXT,
        created_at TEXT NOT NULL DEFAULT (datetime('now'))
    )";

pub const MYSQL_INDEXES: &[&str] = &[];

pub const SQLITE_INDEXES: &[&str] = &[];
